libraries/extractor/src/main/java/androidx/media3/extractor/DtsUtil.java

```diff
@@ -76,6 +76,19 @@ public final class DtsUtil {
         64, 112, 128, 192, 224, 256, 384, 448, 512, 640, 768, 896, 1024, 1152, 1280, 1536, 1920,
         2048, 2304, 2560, 2688, 2816, 2823, 2944, 3072, 3840, 4096, 6144, 7680
       };
+  private static String currentMimeType;
```

Comment:
It's error-prone to set static property on a util class as calling `clearCurrentMimeType` can be easily forgotten. 

It looks like we only need this information in `DefaultAudioTrackBufferSizeProvider`, so I think we have below options:

1. In `DefaultAudioSink.configure`, when calling method [`DefaultAudioTrackBufferSizeProvider.getBufferSizeInBytes`](https://github.com/androidx/media/blob/main/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/audio/DefaultAudioSink.java#L751), pass the encoding as `ENCODING_DTS_HD` if: `outputEncoding == ENCODING_DTS_HD` OR `outputEncoding == ENCODING_DTS && inputFormat.sampleMimeType.contentEquals(MimeTypes.AUDIO_DTS_EXPRESS)`. Note that this "modified" encoding can also flow to the offload branch, if this only applies to passthrough, we also need to check the output mode before passing the "modified" encoding.

2. Add a parameter of `sampleMimeType` to `DefaultAudioTrackBufferSizeProvider.getBufferSizeInBytes`, this requires the customized implementation of `AudioTrackBufferSizeProvider` to change accordingly though.

How do you think of these two options?

Reply:
@tianyif I double checked with my TV SOC colleague and turns out that ENCODING_DTS_HD should be supported by all implementations for DTS Express. So I think there is no need to remap anything. The presence of ENCODING_DTS_HD should be enough to signify that the input stream is actually DTS Express and we can proceed to multiple the buffer size by dtseBufferMultiplicationFactor. I have removed currentMimeType etc in the new commit. Let me know if it is ok. Thanks.

Reply:
Sounds good. Thanks for letting me know!